src/types.ts

```diff
@@ -14,3 +14,8 @@ export interface AudioPlayerContext extends AudioPlayerState {
     player: Howl | null
     load: (args: AudioOptions) => void
 }
+
+export interface AudioPlayerPositionContext {
+    position: number
+    setPosition: (current: number) => void
```

Comment:
if you want to cut down on lines you could ditch your own implementation of `setPosition` in the `usePosition` hook and reuse `setCurrentPosition`. The type here would change to `React.Dispatch<number>` if I'm not mistaken

Reply:
I don't understand what you mean (I'm kinda new to typescript :)).

Reply:
haha that's alright, ill add a suggested change for the TS stuff

Reply:
hmm weird I thought GitHub used to have a feature to suggest changes. Here is what i was thinking:

```typescript
export interface AudioPlayerPositionContext {
  position: number
  setPosition: React.Dispatch<number>
}
```
This way, when you create the context value in the AudioPlayerProvider component you can just place the setter `useState` gives you as the value for `setPosition`.

Reply:
To give you a little more context on why I suggested this: React's typescript definitions export a type for the state setter functions returned from `useState`. It saves you some keystrokes to just use this type rather than implementing your own setPosition method and using its interface as the type on the context's interface, if that makes sense.